src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +139,27 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+function dirTypeScriptFiles(base:string, files?:string[], result?:string[]): string[] {
+    files = files || readdirSync(base) 
+    result = result || [] 
+
+    files.forEach( 
+        function (file) {
+            const newbase = path.join(base, file)
+            if ( statSync(newbase).isDirectory() )
+            {
+                result = dirTypeScriptFiles(newbase, readdirSync(newbase), result)
```

Comment:
this looks like it assumes that `result` should always be overwritten. If that's the case, either:

- assert that `result` is undefined or empty, or
- check for `if(!result)` before this if(), and handle it 

or, do a `result.push(...)` like below.

Reply:
Created a new simplified version that returns immediately if a TS file is found.